dashboard/data/determination.py

```diff
@@ -33,7 +32,23 @@ def find_argument_four_param_logistic(
     :param slope: the steepness of the curve
     :return: argument of the function for given y
     """
-    return ic50 * ((lower_limit - upper_limit) / (y - upper_limit) - 1) ** (1 / slope)
+    x = ic50 * ((lower_limit - upper_limit) / (y - upper_limit) - 1) ** (1 / slope)
+    if type(x) == complex:
+        x = np.nan
+    return x
+
+
+def calculate_modulation_ic50_and_concentration_50(row):
+    modulation_ic50 = four_param_logistic(
+        row["ic50"], row["BOTTOM"], row["TOP"], row["ic50"], row["slope"]
+    )
+    concentration_50 = find_argument_four_param_logistic(
```

Comment:
I think it should be the argument for calculated IC50, not the middle value
(or if it actually is and I misunderstood, I suggest renaming it to `concentration_ic50`)

Reply:
- `ic50` - x (concentration) argument for inflection point
- `modulation_ic50` - activation/inhibition for inflection point i.e. ic50
- ` concentration_50` - x (concentration) for y = 50 (activation/inhibition = 50)

The convention is the same as here:
![obraz](https://github.com/zuzg/drug-screening/assets/82370491/dff22ec8-41a6-422a-8af4-5a85a7bfc708)
Lmk if now's clear 🙏🙏